service/resource/cloudformation/main_stack.go

```diff
@@ -2,19 +2,27 @@ package cloudformation
 
 import (
 	"bytes"
+	"strconv"
 	"text/template"
 
-	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/giantswarm/aws-operator/service/key"
 	"github.com/giantswarm/awstpr"
 )
 
 func newMainStack(customObject awstpr.CustomObject) (StackState, error) {
 	stackName := key.MainStackName(customObject)
+	workers := len(customObject.Spec.AWS.Workers)
+	var imageID string
+	if workers > 0 {
+		imageID = customObject.Spec.AWS.Workers[0].ImageID
+	}
+	clusterVersion := customObject.Spec.Cluster.Version
 
 	mainCF := StackState{
-		Name:    stackName,
-		Outputs: []*awscloudformation.Output{},
+		Name:           stackName,
+		Workers:        strconv.Itoa(workers),
+		ImageID:        imageID,
+		ClusterVersion: clusterVersion,
```

Comment:
IMO it would be good to separate the work of extending the resource state and implementing the update step.

Reply:
Ok, done in #502